### ISSUE TITLE
Bunch of java.lang.NoSuchMethodError exceptions

### ISSUE THREAD

**Issue by mrmitew** [NYTimes/Store#262]
Hello,

First of all, I want to thank you for the great library and thorough documentation! You did an amazing job. Hands down :)

Now to the problem...unfortunately, when I call `.get()` or `.getRefreshing()` on my store, I am getting the following exceptions:

> java.lang.NoSuchMethodError: No static method lambda$apply$1(Lcom/nytimes/android/external/store3/base/impl/RepeatWhenEmits;Lio/reactivex/Observable;)Lio/reactivex/ObservableSource; in class Lcom/nytimes/android/external/store3/base/impl/RepeatWhenEmits; or its super classes (declaration of 'com.nytimes.android.external.store3.base.impl.RepeatWhenEmits' appears in ......

> java.lang.NoSuchMethodError: No static method lambda$lazyCache$0(Lcom/nytimes/android/external/store3/base/impl/RealInternalStore;Ljava/lang/Object;)Lio/reactivex/MaybeSource; in class Lcom/nytimes/android/external/store3/base/impl/RealInternalStore; or its super classes (declaration of 'com.nytimes.android.external.store3.base.impl.RealInternalStore' 

> java.lang.NoSuchMethodError: No static method lambda$lazyCache$0(Lcom/nytimes/android/external/store3/base/impl/RealInternalStore;Ljava/lang/Object;)Lio/reactivex/MaybeSource; in class Lcom/nytimes/android/external/store3/base/impl/RealInternalStore; or its super classes (declaration of 'com.nytimes.android.external.store3.base.impl.RealInternalStore'

Here's my gradle configuration:
```
implementation "io.reactivex.rxjava2:rxjava:2.0.7"
implementation "io.reactivex.rxjava2:rxandroid:2.0.1"
implementation 'com.nytimes.android:cache3:3.0.0-
implementation 'com.nytimes.android:store3:3.0.0-beta'
implementation 'com.nytimes.android:middleware3:3.0.0-beta'
implementation 'com.nytimes.android:filesystem3:3.0.0-beta'
```

**Comment by digitalbuddha** [comment]
Let me guess. You're using the new built in desugar?

**Comment by mrmitew** [comment]
Yeah, I am using the java 8 lang features without retrolambda. Is that the problem?

**Comment by digitalbuddha** [comment]
Yup! I've reported the bug to Google. There was a bug in the built in lambda desugar in android gradle tools 3.0. If you add Retrolambda it works fine. Google will have a fix out in next few versions https://issuetracker.google.com/issues/62456849

**Comment by mrmitew** [comment]
That totally fixed the problem, indeed! Thanks @digitalbuddha. I will stick with Retrolambda until the bug fix  is released then. Thanks a bunch! Probably a good idea will be to mention about this problem in the repo
s README for other people like me?

*EDIT*: Is there any other reason you are still sticking with Retrolambda, by the way?

**Comment by digitalbuddha** [comment]
Nope! Feel free to pull it out. But since old artifacts still had it figured we need the fix from Google anyway

**Comment by mrmitew** [comment]
Gotcha! Looks like we should expect the fix in the next alpha release (3.0.0-alpha10).

PS: Thank you for the quick responses!
